client/state/document-head/selectors.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { getSelectedSite, getGroupName } from 'state/ui/selectors';
+
+/**
+ * Returns the document title as set by the DocumentHead component or setTitle
+ * action.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Document title
+ */
+export function getTitle( state ) {
+	return state.documentHead.title;
+}
+
+/**
+ * Returns a count reflecting unread items.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Unread count (string because it can be e.g. '40+')
+ */
+export function getUnreadCount( state ) {
+	return state.documentHead.unreadCount;
+}
+
+/**
+ * Returns the formatted document title, based on the currently set title,
+ * unreadCount, and selected site.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Formatted title
+ */
+export function getFormattedTitle( state ) {
+	const siteSpecificGroups = [ 'sites', 'editor' ];
+	const title = getTitle( state );
+	const unreadCount = getUnreadCount( state );
+	const site = getSelectedSite( state );
+
+	let pageTitle = '';
+
+	if ( unreadCount ) {
+		pageTitle += '(' + unreadCount + ') ';
+	}
+
+	pageTitle += title;
+
+	// Display site name as title part only if we're in 'My Sites'
+	if ( includes( siteSpecificGroups, getGroupName( state ) ) && site ) {
+		pageTitle = appendSite( pageTitle, site );
+	}
+
+	if ( pageTitle ) {
+		pageTitle = decodeEntities( pageTitle ) + ' — WordPress.com';
+	} else {
+		pageTitle = 'WordPress.com';
+	}
+
+	return pageTitle;
+}
+
+function appendSite( title, site ) {
```

Comment:
> - I'd rather this be built into `getFormattedTitle` since it's (a) not a selector on its own and (b) is not used elsewhere anyways

Yep, sounds good.

> - Is there something achieved by the condition in the site condition that's not reflected by the [logic for computed `title`](https://github.com/Automattic/wp-calypso/blob/f2194b6dd24929e1e5be0ee69e0c800612cafd06/client/lib/site/computed-attributes.js#L26) (soon to be a native state selector in #7322)? 

No, I think we can perfectly re-use that. Care to merge #7322 (which has green light) so I can use that selector?